utils.c

```diff
@@ -262,7 +262,7 @@ fdwalk (int proc_fd, int (*cb)(void *data, int fd), void *data)
   int res = 0;
   DIR *d;
 
-  dfd = openat (proc_fd, "self/fd", O_DIRECTORY | O_PATH);
+  dfd = openat (proc_fd, "self/fd", O_DIRECTORY);
```

Comment:
As long as we're changing the flags, it seems worth canonicalizing with other projects.  libglnx uses:

```
  int flags = O_RDONLY | O_NONBLOCK | O_DIRECTORY | O_CLOEXEC | O_NOCTTY;
  if (!follow)
    flags |= O_NOFOLLOW;
```

Which I think was taken from systemd.